### ISSUE TITLE
Check and warn for Juju and Snap packaging that is on latest/stable channels

### ISSUE THREAD

**Issue by dwbfox** [canonical/hotsos#721]
**TL;DR**
The request here is for hotsos to be able to detect Charms in sosreports that are on `latest/stable` and provide a warning/advisory message, as ideally users should be pinned to specific versions/releases for both Juju charms and Snap packages for production workloads.

**Juju Charms**
Currently, there's an effort to deprecate the release of packages on`latest/stable` channels in Juju charms, specifically Openstack related charms. This could cause issues for existing deployments whose bundle may still be keyed to use `latest/stable` instead of a pinned version of a charm, thus cutting them off from future updates and releases unless the Charm is manually refreshed to a new channel.

The caveat is that not all published charms follow this new convention (that is avoiding the use of `latest/stable`) and this change in process seems to be limited to Openstack related charms, which could be problematic when implementing this kind of check. 

**Snap packages**
Snap packages also fall into this problem. Depending on the snap maintainer, some projects will have a concept of versioned channels for "stable" releases while using `latest/stable` as a trunk/master branch where the newest code, breaking changes, and features are added. The check here would be to warn when certain production related packages are on `latest/stable` 

For example, LXD team has a concept of "LTS" releases for major versions ending in x.0 -- so for example the current LTS release of LXD snap is `5.0`, and the next one is `6.0`. But there are intermediate releases in between that may contain schema changes, new features, and so on. Thus the request here would be, at least for LXD, to check for these non-LTS versions of the snap installed and provide a warning for it.

However this wouldn't pan out very well for all snap packages as snap by default will auto-install the `latest/stable` channel, so doing a blanket check on all snaps to see if they're on `latest/stable` would be too noisy:

```
snap list | grep stable -c
29
```
